src/main/java/org/example/printer/Printer.java

```diff
@@ -0,0 +1,39 @@
+package org.example.printer;
+
+public class Printer {
+    private int tonerLevel;
+    private int pagesPrinted;
+    private boolean duplex;
+    public Printer(int tonerLevel, boolean duplex) {
+        if (tonerLevel > -1 && tonerLevel <= 100) {
+            this.tonerLevel = tonerLevel;
+        } else {
+            this.tonerLevel = -1;
+        }
+        this.duplex = duplex;
+        this.pagesPrinted = 0;
+    }
+    public int addToner(int tonerAmount) {
+        if (tonerAmount > 0 && tonerAmount <= 100) {
+            if (this.tonerLevel + tonerAmount > 100) {
+                return -1;
+            } else {
+                this.tonerLevel += tonerAmount;
+                return this.tonerLevel;
+            }
+        } else {
+            return -1;
+        }
+    }
+    public int printPages(int pages) {
+        int pagesToPrint = pages;
+        if (this.duplex) {
+            pagesToPrint = (pages + 1) / 2;
```

Comment:
this calculation is not correct, think how to process even and odd numbers

Reply:
Maybe this correct calculation pagesToPrint = pages / 2 ?

Reply:
not exactly. think about processing odd and even numbers. If you pass 2,4,6,8 `pages/2` is ok, but how to process 3,5,7... ?

Reply:
Then I think it can be defined as follows  
(pages%2)